torch/_inductor/codegen/cpp.py

```diff
@@ -2094,6 +2107,8 @@ def get_reduction_code_buffer(loops, buffer="prefix"):
                             prefix = kernel.reduction_prefix
                             if loop.parallel:
                                 prefix = prefix + kernel.parallel_reduction_prefix
+                            else:
+                                prefix = prefix + kernel.reduction_weight_recps
```

Comment:
why can't we reuse the `kernel.reduction_prefix` here but have to create a new `kernel.reduction_weight_recps`?

Reply:
Can't reuse `kernel.reduction_prefix` directly here, otherwise there will be an unused `weight_recps` at the beginning in the parallel reduction case.

Reply:
ok, how about name it `non_parallel_reduction_prefix` as the counterpart of `parallel_reduction_prefix`? The weight reciprocals are one of such prefix.